service-map/src/main/java/fi/nls/oskari/map/geometry/ProjectionHelper.java

```diff
@@ -280,4 +283,36 @@ public static void swapGeometryXY(FeatureCollection<SimpleFeatureType, SimpleFea
             featuresIter.close();
         }
     }
+
+    /**
+     *  Transforms all geometries in Featurecollection
+     *  REMARK crs (sourceCRS) must be defined in feature default geometrydescriptor
+     *  (fc.getSchema().getGeometryDescriptor().getCoordinateReferenceSystem() must be not null
+     * @param featureCollection
+     * @param sourceCrs
+     * @param targetCrs
+     * @return
+     */
+    public static FeatureCollection<SimpleFeatureType, SimpleFeature> transformFeatureCollection(FeatureCollection<SimpleFeatureType,
+            SimpleFeature> featureCollection, String sourceCrs, String targetCrs) {
+
+        if(sourceCrs.toUpperCase().equals(targetCrs.toUpperCase())){
+            return featureCollection;
+        }
+
+        try {
+            CoordinateReferenceSystem crsSource = CRS.decode(sourceCrs, true);
+            CoordinateReferenceSystem crsTarget = CRS.decode(targetCrs, true);
+            Query query = new Query();
+            query.setCoordinateSystem(crsSource);
+            query.setCoordinateSystemReproject(crsTarget);
+            SimpleFeatureSource source = DataUtilities.source(featureCollection);
+            return source.getFeatures(query);
+        }
+        catch (Exception e) {
+            log.error(e, "Transform featureCollection geometries failed! Params:: sourceSRS: ", sourceCrs, "targetSRS: ", targetCrs);
+        }
+        return featureCollection;
```

Comment:
Actually the other methods in the class return null in this case so this should do the same